packages/base-driver/lib/basedriver/capabilities.js

```diff
@@ -336,43 +335,74 @@ function processCapabilities(
 
 /**
  * Return a copy of a capabilities object which has taken everything within the 'options'
- * capability and promoted it to the top level. Note that this function is assumed to be run after
- * all vendor prefixes have already been stripped from the top level. So we are dealing with e.g.
- * 'options' and not 'appium:options' at this point. Any prefixes _inside_ the 'options' capability
- * will themselves be stripped. This is designed as an internal function, not one to operate on
- * user-constructed capabilities.
+ * capability and promoted it to the top level.
  *
- * @param {object} originalCaps - the capabilities to analyze and promote from 'options'
- * @return {object!} - the capabilities with 'options' promoted if necessary
+ * @template {Constraints} C
+ * @param {import('@appium/types').W3CCapabilities<C>} originalCaps
+ * @return {import('@appium/types').W3CCapabilities<C>} the capabilities with 'options' promoted if necessary
  */
 function promoteAppiumOptions(originalCaps) {
-  const appiumOptions = originalCaps[APPIUM_OPTS_CAP];
-  if (!appiumOptions) {
-    return originalCaps;
-  }
+  const promoteForObject = (obj) => {
+    const appiumOptions = obj[PREFIXED_APPIUM_OPTS_CAP];
+    if (!appiumOptions) {
+      return obj;
+    }
 
-  let caps = _.cloneDeep(originalCaps);
-  if (!_.isPlainObject(appiumOptions)) {
-    throw new errors.SessionNotCreatedError(`The ${APPIUM_OPTS_CAP} capability must be an object`);
-  }
+    if (!_.isPlainObject(appiumOptions)) {
+      throw new errors.SessionNotCreatedError(`The ${PREFIXED_APPIUM_OPTS_CAP} capability must be an object`);
+    }
+    if (_.isEmpty(appiumOptions)) {
+      return obj;
+    }
 
-  // first get rid of any prefixes inside appium:options
-  const strippedAppiumOptions = stripAppiumPrefixes(appiumOptions);
-  // warn if we are going to overwrite any keys on the base caps object
-  const overwrittenKeys = _.intersection(Object.keys(caps), Object.keys(strippedAppiumOptions));
-  if (overwrittenKeys.length > 0) {
-    log.warn(
-      `Found capabilities inside ${PREFIXED_APPIUM_OPTS_CAP} that will overwrite ` +
-        `capabilities at the top level: ${JSON.stringify(overwrittenKeys)}`
+    log.debug(
+      `Found ${PREFIXED_APPIUM_OPTS_CAP} capability present; will promote items inside to caps`
     );
-  }
 
-  // now just apply them to the main caps object
-  caps = {...caps, ...strippedAppiumOptions};
+    const shouldAddVendorPrefix = (/** @type {string} */ capName) => !capName.startsWith(APPIUM_VENDOR_PREFIX);
+    const verifyIfAcceptable = (capName) => {
+      if (!_.isString(capName)) {
+        throw new errors.SessionNotCreatedError(
```

Comment:
If we have a non-array object, then assuming `capName` is a property of that object, it _must_ be a `string`, because a `symbol` cannot be transmitted over the wire.  Right?

Reply:
in general yes. just being a bit paranoidal